### DISCUSSION TITLE
mysql engine / HowTo

### DISCUSSION THREAD

**Discussion by 120L020910fanshuo** [searxng/searxng#3923] [Q&A]
![image](https://github.com/user-attachments/assets/30010b34-d52c-40d0-8639-2ff1dee6164f)
![image](https://github.com/user-attachments/assets/4ee71dba-3fef-49b3-8e0d-90237b89072a)
Here are my setting.yml and mysql database.And there are not other changes.
What causes this problem?
![image](https://github.com/user-attachments/assets/b033a117-0ba0-482d-aff4-0bdf8461b6f0)
![image](https://github.com/user-attachments/assets/09a55b63-fd05-431a-bf24-72878a2f118e)
Thank you!

**Comment by 120L020910fanshuo** [comment]
In other words, do I just need to modify this part in the settings.yml to add my own MySQL database as a search source, without changing any other files?Welcome, experts, for your guidance.

**Reply by return42** [reply]
In your /etc/searxng/settings.yml you need an entry like:

https://github.com/searxng/searxng/blob/2986681b31a60d67e0468affb257a52246220f11/searx/settings.yml#L2033-L2041

You need to define the DB connection (database, username & password) and a query term (query_str).  A more elaborate example can be found in the docs: https://docs.searxng.org/dev/engines/offline/sql-engines.html#engine-sqlite

**Reply by 120L020910fanshuo** [reply]
I'm very sorry that it took me so long to see your reply. I modified settings.yml according to the technical documentation you provided and changed it to the following simple settings.
![image](https://github.com/user-attachments/assets/574638f8-2a68-43c2-8500-f2c44c0a1978)
My database is as follows:
![image](https://github.com/user-attachments/assets/755aa91d-39f2-41a3-8bd7-4d4b1990a071)
Then I rebuilt the image and pushed it to my Docker Hub. Image: docker.io/bianbao/se:v12.
But it is still the following result
![image](https://github.com/user-attachments/assets/225465c2-16ce-4974-b7b2-74c58971bb2d)
![image](https://github.com/user-attachments/assets/f8243d81-b07a-4a34-9e1e-ff8c4119bf91)
I may be a bit stupid. Thank you for your guidance.

**Reply by 120L020910fanshuo** [reply]
Sorry, I unmarked the answer in case you can't see my reply. Sorry.

**Reply by 120L020910fanshuo** [reply]
At the same time, I also tried the SQLite database. I downloaded the movie database in the technical document and put it in searx/data. I directly used the default code in setting.yml and then performed a docker build to build an image. However, when I used my own image, the following problem still occurred. I am very sorry to cause you so much trouble, but I hope to get your help.
![image](https://github.com/user-attachments/assets/6af02b4e-851b-4f1c-8a51-32f5ffef040f)
The following picture shows a problem similar to MySQL.
![image](https://github.com/user-attachments/assets/511bdd7c-0a21-44ce-b81c-656a3d3f2736)

**Reply by return42** [reply]
Sorry, I can't reproduce your issue:

```
$ cd searx/data/
$ curl "https://liste.mediathekview.de/filmliste-v2.db.bz2" -o filmliste-v2.db.bz2
$ bunzip2 filmliste-v2.db.bz2 
```

and in the settings ..

```diff
diff --git a/searx/settings.yml b/searx/settings.yml
index 131ee741c..654e1f3ca 100644
--- a/searx/settings.yml
+++ b/searx/settings.yml
@@ -1727,20 +1727,20 @@ engines:
   #   https://liste.mediathekview.de/filmliste-v2.db.bz2
   # and unpack into searx/data/filmliste-v2.db
   # Query to test: "!demo concert"
-  #
-  # - name: demo
-  #   engine: sqlite
-  #   shortcut: demo
-  #   categories: general
-  #   result_template: default.html
-  #   database: searx/data/filmliste-v2.db
-  #   query_str:  >-
-  #     SELECT title || ' (' || time(duration, 'unixepoch') || ')' AS title,
-  #            COALESCE( NULLIF(url_video_hd,''), NULLIF(url_video_sd,''), url_video) AS url,
-  #            description AS content
-  #       FROM film
-  #      WHERE title LIKE :wildcard OR description LIKE :wildcard
-  #      ORDER BY duration DESC
+
+  - name: demo
+    engine: sqlite
+    shortcut: demo
+    categories: general
+    result_template: default.html
+    database: searx/data/filmliste-v2.db
+    query_str:  >-
+      SELECT title || ' (' || time(duration, 'unixepoch') || ')' AS title,
+             COALESCE( NULLIF(url_video_hd,''), NULLIF(url_video_sd,''), url_video) AS url,
+             description AS content
+        FROM film
+       WHERE title LIKE :wildcard OR description LIKE :wildcard
+       ORDER BY duration DESC
 
   - name: tagesschau
     engine: tagesschau
```

then `make run` and in the WEB browser search for [!demo concert](http://127.0.0.1:8888/searx/search?q=%21demo+concert)

**Reply by 120L020910fanshuo** [reply]
Thank you! The “make run” part has been successful. 
However, during the process of “Build the image”,
 I first performed “git clone https://github.com/searxng/searxng.git”,
 then modified settings.yml and added searx/data/filmliste-v2.db, and then performed “make docker.build”. 
The results are as follows: 

bianbao@bianbao-virtual-machine:~/demo/searxng$ docker images
REPOSITORY TAG IMAGE ID CREATED SIZE
searxng/searxng 2024.10.19-050451347 935fcc9bb94a 34 minutes ago 1.39GB
searxng/searxng latest 935fcc9bb94a 34 minutes ago 1.39GB
alpine 3.20 91ef0af61f39 6 weeks ago 7.8MB. 

Then rename searxng/searxng:latest as bianbao/se:v19.
In another project, this part uses the searxng image developed by you. After I replace it with the image generated by myself, the demo still doesn't work.
Before：
![image](https://github.com/user-attachments/assets/33488132-b04a-4359-a89e-4fcee3aefb74)
After：   
![image](https://github.com/user-attachments/assets/b1af45f4-a1c5-4367-b982-874d1d715f62)
![image](https://github.com/user-attachments/assets/f603d03e-a0ec-46b0-8895-299305a556de)
I don't know if there's an error when adding the demo engine and generating the image. Sorry to trouble you. I'd be extremely grateful if you could guide me again.

**Reply by return42** [reply]
> After I replace it with the image generated by myself, the demo still doesn't work.

Here at https://github.com/searxng/searxng/ we discuss issues related to the SearXNG core.  To me it seems you have some trouble with your docker workflow / such topics might be better discussed in the community on Matrix [#searxng:matrix.org](https://matrix.to/#/#searxng:matrix.org).
